webapp/src/main/java/com/box/l10n/mojito/okapi/ExtractUsagesFromTextUnitComments.java

```diff
@@ -0,0 +1,105 @@
+package com.box.l10n.mojito.okapi;
+
+import com.box.l10n.mojito.okapi.filters.UsagesAnnotation;
+import net.sf.okapi.common.resource.TextUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ *
+ * @author emagalindan
+ */
+@Component
+public class ExtractUsagesFromTextUnitComments {
+
+    /**
+     * logger
+     */
+    static Logger logger = LoggerFactory.getLogger(ExtractUsagesFromTextUnitComments.class);
+
+    @Autowired
+    TextUnitUtils textUnitUtils;
+
+    private static final String USAGES_PATTERN = "\\s*?<locations>\n?(?<usages>(.*?\\s)*?)</locations>";
+    private static final String USAGES_GROUP_NAME = "usages";
+
+    /**
+     * Add usage locations to the text unit
+     *
+     * @param textUnit the text unit to add usages
+     */
+    public void addUsagesToTextUnit(TextUnit textUnit) {
+        Set<String> usagesFromComment = extractUsagesFromTextUnit(textUnit);
+        textUnit.setAnnotation(new UsagesAnnotation(usagesFromComment));
+    }
+
+    /**
+     * Get and delete the usage locations from the comment in textUnit.
+     *
+     * @param textUnit to get the usages from
+     * @return the locations or empty set
+     */
+    protected Set<String> extractUsagesFromTextUnit(TextUnit textUnit) {
```

Comment:
let's use the same method scope for all method, I like "package-protected" those days ... which is no keyword

`protected Set<String> extractUsagesFromTextUnit(TextUnit textUnit) {` -->` Set<String> extractUsagesFromTextUnit(TextUnit textUnit) {`